src/components/GameDetail/GameProgress.vue

```diff
@@ -0,0 +1,279 @@
+<template lang="html">
+  <div :class="['game-progresses', { small }]">
+    <div v-if="hasProgress && !showProgressField" class="progress">
+      <div class="progress-data">
+        <progress
+          class="progress-bar"
+          max="100"
+          :value="localProgress.number"
+        >
+          <div class="progress-bar-fallback">
+            <span :style="style">
+              Progress: {{localProgress.number}}%
+            </span>
+          </div>
+        </progress>
+        <div class="progress-bar-label">
+          {{localProgress.number}}%
+        </div>
+      </div>
+
+      <button
+        v-if="!small"
+        class="primary"
+        @click="editProgress"
+      >
+        Edit progress
+      </button>
+    </div>
+
+    <div
+      class="progress-field"
+      v-if="showProgressField && !small"
+    >
+      <input
+        v-model.number="localProgress.number"
+        placeholder="50"
+        type="number"
+        autofocus
+      >
+
+      <div class="progress-action">
+        <div class="progress-edit">
+          <button class="secondary" @click="reset">
+            {{ $t('global.cancel') }}
+          </button>
+
+          <button class="primary" @click="saveProgress">
+            {{ $t('global.save') }}
+          </button>
+        </div>
+
+        <button class="danger" @click="deleteProgress">
+          Delete progress
+        </button>
+      </div>
+    </div>
+
+    <button
+      v-if="!hasProgress &&
+            !showProgressField &&
+            !small"
+      class="primary"
+      @click="addProgress"
+    >
+      <i class="fas fa-clock" />
+      {{ $t('progresses.addProgress') }}
+    </button>
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from 'vuex';
+
+export default {
+  props: {
+    small: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      showProgressField: false,
+      localProgress: {
+        number: null,
+      },
+    };
+  },
+
+  computed: {
+    ...mapState(['game', 'progresses']),
+    ...mapGetters(['gameProgress']),
+
+    hasProgress() {
+      return this.gameProgress && this.gameProgress.number;
+    },
+
+    style() {
+      return `width: ${this.localProgress.number}%`;
+    },
+  },
+
+  mounted() {
+    this.reset();
+  },
+
+  methods: {
+    reset() {
```

Comment:
I reached the end of my abilities to understand Vue at this point.

Sometimes the progress shows, often times it doesn't. If it does and I close the modal or open a modal it disappears again and then it won't come back.

However, I noticed the following:
- when I do `console.log(this)` at the beginning of `reset()` I will get an empty `gameProgress`
- when I open a game's modal and look at the log it shows everything I've saved in `gameProgress` but the progress bar will still be empty

So, I'm guessing that the something is missing and I can't figure out what.

Reply:
I think it's because in the getter you added you are using `game` from state, but `game` is used to store the active game (game viewed in modal). So your getter only gets the data needed when the modal is active.

Reply:
I'd be happy to help you get through the finish line, lmk if you want me to make some adjustments, looking great so far!

Reply:
Please do.

I was trying to bastardize the Tags modules all day and just made it worse. I could use the help!